src/client/core/utils/wait-for.js

```diff
@@ -0,0 +1,30 @@
+import hammerhead from '../deps/hammerhead';
+
+var Promise = hammerhead.Promise;
+
+
+export default function (fn, delay, timeout) {
+    return new Promise((resolve, reject) => {
+        var result = fn();
+
+        if (result) {
+            resolve(result);
+            return;
+        }
+
+        var intervalId = window.setInterval(() => {
```

Comment:
Shouldn't we use native methods like here: https://github.com/AlexanderMoskovkin/testcafe-phoenix/blob/client-driver/src/client/core/utils/delay.js#L8 ?